backend/main.py

```diff
@@ -1,14 +1,33 @@
 # Simple backend server using flask framework
 
-from flask import Flask
+from flask import Flask, jsonify, request
 from flask_cors import CORS, cross_origin
 
 app = Flask(__name__)
 CORS(app)
 
+'''
+    Get sample data from local file with a fixed output lines
+    TODO: too naive, improvement required.
+'''
+@app.route('/data')
+def getData():
+    fn = 'Power_sample_data.csv'
+    numLines = 7200000  # assume number of records is accessible upon deployment
+    maxLines = request.args.get('number', default=6000, type=int)
+    frequency = numLines / maxLines
+    data = list()
+    with open(fn, 'r') as fr:
+        for i, line in enumerate(fr):
+            if i % frequency == 0:
+                data.append(line.strip('\n').split(','))
```

Comment:
Looks this is a interface to render data to frontend.
Better move the logic which process the raw data in a separate function.
e.g. 
def downsample_raw_data(file_name):

The best practice is to make a function only do one job. And it should be good enough to understand the function from the name without reading the detailed implementation.